prow/config.yaml

```diff
@@ -155,6 +155,7 @@ branch-protection:
               - test-infra-test-jobs-yaml-definitions
               - test-infra-validate-scripts
               - test-infra-validate-configs
+              - test-infra-governance
```

Comment:
It looks like it is not needed. Branch Protector defines all jobs as required by default. See #304